### PR TITLE
Node Cloud plugin class generator

### LINKED ISSUE #52: Automating the service class creation process of Node Cloud plugins
### Describe the solution you'd like
 * Automate the process of manually creating classes for services in the Node Cloud plugins
 * Data needs to be extracted from the cloud provider SDKs and a parser needs to be used in this process
 * After extracting data JavaScript classes can be automatically generated using a transformer
### Impact
* A new component to the project which can automatically generate JavaScript classes

### PR DESCRIPTION
**Pull request by rajitha1998**
Fixes #52  

## Proposed Changes
* This pull request aims to automate Node Cloud class generation 

* To execute the generator navigate to generator directory and run  
`tsc main.ts && node main.js`

### CODE REVIEW

**Comment by rajikaimal** [comment]
Let's add this to different branch, `nodecloud-next`

File: classGenerator/generator/transformer.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import * as ts from "typescript";
```
Comment (rajikaimal) [inline classGenerator/generator/transformer.ts]: Can we use packages for parser and transformer?

Comment (rajitha1998) [reply classGenerator/generator/transformer.ts]: * I updated the `generator.ts, paser.ts ` files by only importing the relevant functions/objects from the typescript package 😀 
* In the` transformer.ts` file, we are using the following functions/ objects from the typescript package. Do we need to import them separately?
```
Node, TransformationContext,  SourceFile,  EmitHint, ScriptTarget, Printer,
NewLineKind, createPrinter, createSourceFile, updateIdentifier, updateClassDeclaration,
createNodeArray, createIdentifier, visitEachChild, visitNode, transform
```
@rajikaimal

**Comment by rajitha1998** [comment]
> Let's add this to different branch, `nodecloud-next`

👍, I think I don't have access to create a branch in this repo. Can you help me with that?

**Comment by rajikaimal** [comment]
Done, https://github.com/leopardslab/nodecloud/tree/nodecloud-next

**Comment by rajitha1998** [comment]
> Done, https://github.com/leopardslab/nodecloud/tree/nodecloud-next

Thank you :)

**Comment by rajitha1998** [comment]
Closing this as the changes went with #51

**Pull request closed without merging**